Improve README.md (#107)
NVIDIA/k8s-dra-driver: README.md

   can be done by following the instructions
   [here](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html).
1. Configure the NVIDIA Container Runtime as the **default** Docker runtime:
   ```console
   sudo nvidia-ctk runtime configure --runtime=docker --set-as-default
   ```
1. Restart Docker to apply the changes:
   ```console
   sudo systemctl restart docker
   ```
1. Set the `accept-nvidia-visible-devices-as-volume-mounts` option to `true` in
   the `/etc/nvidia-container-runtime/config.toml` file to configure the NVIDIA
   Container Runtime to use volume mounts to select devices to inject into a
   container.

> 

Reviewer: While you're changing things do you want to add this command below this step as well:
```
sudo nvidia-ctk config --set accept-nvidia-visible-devices-as-volume-mounts=true --in-place
```

   Container Runtime to use volume mounts to select devices to inject into a
   container.

> 1. Show the current GPU information of the machine

Reviewer 2: I think this step is covered at  https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html)

Author: Yes, but would it be helpful to have it in the root README. For example, I was following the instructions in this README in my first attempt.

> 1. Show the current GPU information of the machine

Reviewer: ```suggestion
1. Show the current set of GPUs on the machine
```

   ```bash
   nvidia-smi -L
   ```

We start by first cloning this repository and `cd`ing into it.
All of the scripts and example Pod specs used in this demo are in the `demo`
subdirectory, so take a moment to browse through the various files and see
what's available:

> ```bash

Reviewer 2: I see bash/console should we agree on a single one? is there a benefit of having one vs the other on specific steps? (is one better for copy/pasting than the other for example?

Author: Changed all of them to `console`.

This also makes the built images available to the `kind` cluster.

We now install the NVIDIA GPU DRA driver:

> ```bash

Author: It's inherited from the original script. Let's use console, which I believe it's what's used in k8s.

> ```bash

Reviewer 2: https://github.com/NVIDIA/k8s-dra-driver/pull/107#discussion_r1591326785

./demo/clusters/kind/install-dra-driver.sh
```

This should show two pods running in the `nvidia-dra-driver` namespace:
```console
$ kubectl get pods -n nvidia-dra-driver
```

> ```

Reviewer: What is the purpose of separating this if you keep the `$`?

Author: A user can copy/paste and run the command directly.

Author: A user can copy/paste and run an example command directly.

Reviewer: I'm trying to say that you didn't remove the `$` in your diff.

Author: Removed it.

### Run the examples by following the steps in the demo script
Finally, you can run the various examples contained in the `demo/specs/quickstart` folder.
The `README` in that directory shows the full script of the demo you can walk through.

```bash
cat demo/specs/quickstart/README.md
```

> Depending on the GPU resources in the cluster, running the first three examples will produce output similar to the following:

Reviewer: ```suggestion
Depending on which GPUs are available, running the first three examples will produce output similar to the following:
```

```console
kubectl apply --filename=demo/specs/quickstart/gpu-test{1,2,3}.yaml
```

```console

> kubectl get pod -A -l app=pod

Reviewer: This does read well after this change. Can you update the text to first describe running the `apply` call and then running `kubectl` get to see them starting up.

Author: Updated it.

```
```
NAMESPACE           NAME                                       READY   STATUS    RESTARTS   AGE
gpu-test1           pod1                                       1/1     Running   0          34s
gpu-test1           pod2                                       1/1     Running   0          34s
gpu-test2           pod                                        2/2     Running   0          34s
gpu-test3           pod1                                       1/1     Running   0          34s
gpu-test3           pod2                                       1/1     Running   0          34s
```
```console
kubectl logs -n gpu-test1 -l app=pod
```

> ```

Reviewer: I looked at the generated README with this change, and I prefer the look of the original. Is the purpose here to make it easier to copy-and-paste the command where the output is attached?

Author: Yes, otherwise, a user has to manually delete the character `$`. Also some examples contain the output of a command in the console quote, which makes the command invalid.

Running
```

> ./demo/clusters/kind/delete-cluster.sh

Reviewer: wrap in `console`?
